sdcm/utils/raft/common.py

```diff
@@ -139,18 +191,65 @@ def clean_and_restart_bootstrap_after_abort(self):
         self.bootstrap_node.stop_scylla_server(ignore_status=True, timeout=600)
         # Clean garbage from group 0 and scylla data and restart setup
         try:
-            if self.verification_node.raft.get_diff_group0_token_ring_members():
+            if self.verification_node.raft.get_diff_group0_token_ring_members() or \
+                    self.verification_node.raft.get_group0_non_voters():
                 self.verification_node.raft.clean_group0_garbage(raise_exception=True)
+
+            if not self.is_bootstrapped_successfully():
                 LOGGER.debug("Clean old scylla data and restart scylla service")
                 self.bootstrap_node.clean_scylla_data()
-            with ignore_raft_topology_cmd_failing(), ignore_raft_transport_failing(), \
-                    adaptive_timeout(operation=Operations.NEW_NODE, node=self.verification_node, timeout=3600) as bootstrap_timeout:
 
-                self.bootstrap_node.start_scylla_server(verify_up_timeout=bootstrap_timeout, verify_down=True)
-                self.bootstrap_node.start_scylla_jmx()
-            self.db_cluster.check_nodes_up_and_normal(
-                nodes=[self.bootstrap_node], verification_node=self.verification_node)
+            watcher_startup_failed = partial(self.watch_startup_failed, timeout=3600)
+            LOGGER.debug("Start rebootstrap as new node")
+            ParallelObject(objects=[self._rebootstrap_node, watcher_startup_failed], timeout=3800).call_objects()
+            LOGGER.debug("Node is up")
+        except NodeSetupFailed as exc:
+            LOGGER.error("Scylla service restart failed: %s", exc)
+            self.clean_unbootstrapped_node()
+            raise BootstrapStreamErrorFailure(f"Rebootstrap failed with error: {exc}") from exc
+        except ExitByEventError as exc:
+            LOGGER.error("Event stopped: %s", exc)
+            if self.bootstrap_node.db_up():
+                LOGGER.info("Node is up")
+            else:
+                LOGGER.info("Clean node")
+                self.clean_unbootstrapped_node()
+
         except Exception as exc:  # pylint: disable=broad-except  # noqa: BLE001
             LOGGER.error("Scylla service restart failed: %s", exc)
             self.clean_unbootstrapped_node()
             raise BootstrapStreamErrorFailure(f"Rebootstrap failed with error: {exc}") from exc
+        finally:
+            self.bootstrap_node.stop_wait_db_up_event.clear()
+
+
+class MonitorDecommissionOperation:
+    DECOMMISSION_STATE = "DECOMMISSIONING"
+
+    def __init__(self, target_node: BaseNode, timeout=7200):
+        self.timeout = timeout
+        self.target_node = target_node
+        self.db_cluster: BaseScyllaCluster = target_node.parent_cluster
+        self.target_node_ip = target_node.ip_address
+
+    @property
+    def verification_node(self) -> BaseNode:
+        return random.choice(self.db_cluster.get_nodes_up_and_normal())
```

Comment:
not recommended, any of those nodes might be participating in a nemesis.

I would recommend using `Nemesis.run_nemesis` context manager for getting a node which isn't part of nemesis right now.

Reply:
fixed